src/geometries/ParametricGeometry.js

```diff
@@ -74,28 +78,36 @@ function ParametricBufferGeometry( func, slices, stacks ) {
 
 			var u = j / slices;
 
-			var p = func( u, v );
-			vertices.push( p.x, p.y, p.z );
+			// vertex
+
+			p0 = func( u, v, p0 );
```

Comment:
Writing it like so is sufficient:
```
func( u, v, p0 );
```
Personal preference, of course. I prefer it because it makes it very clear the memory is pre-allocated.